python-package/xgboost/sklearn.py

```diff
@@ -61,14 +61,17 @@ class XGBModel(XGBModelBase):
     learning_rate : float
         Boosting learning rate (xgb's "eta")
     n_estimators : int
-        Number of boosted trees to fit.
+        Number of trees to fit. If the mode is gbm, this is the number of boosted rounds.
```

Comment:
I don't understand why do we need to split it into two modes?  In my conceptual model,  XGBoost is a boosting framework, `tree`, `linear` and `random forest` are different base models of `XGBoost` that being boosted.  Adding `XGBoostRF*` makes sense to me since it creates a better view of different base models, we can apply the same to `linear` model and `tree` model so we have `XGBoostLinear*` and `XGBoostTree*`.  But I want to keep the semantic of parameters like `n_estimators` unchanged and mean the same thing in different contexts.  We can declare `n_estimator` exclusively in `XGBoostRF*` so that other two interfaces won't accept this parameters accidentally.

I'm open to discussions on this.

Reply:
`n_estimators` always has the meaning of the number of simple classifiers used (in this case, trees). The only difference is in how they are trained.

I removed the comment regarding the mode, as it no longer applies.